### ISSUE TITLE
RGSoC questions

### ISSUE THREAD

**Issue by mweitzel** [mweitzel/firetower#9]
If you have questions or comments pertaining to [Rails Girls Summer of Code](https://teams.railsgirlssummerofcode.org/projects/131-firetower), please discuss here! Many of you have similar questions, and I'd much prefer to answer them openly where other's could benefit from them.

A few questions so far:

- _could provide me with some more information about this project_

I'm happy to, but please be specific. If you'd like to see something in the documentation, whether its usage or internal mechanics, please submit an issue!

- _I'd like to contribute beforehand, what should I work on?_

I am open to pull requests on [any issue](https://github.com/mweitzel/firetower/issues). If there is anything you'd like to see in this project, whether its a feature, better documentation, or resources for learning unix fundamentals, create an issue.

That said, this is a fairly small project, and it is possible to clear all the tickets out before summer even starts, which would be a little unfortunate. For the a RGSoC project, this makes most sense for a team with little experience with unix who wants the exposure and a deeper understanding of bash. I'd coding tickets stay available to them. PRs for the documentation, are more than welcome. Right now there is no contributing guide, which I'm learning would be very helpful to have before summer.

- _I want to contribute to this project in the summer, what should I work on in the mean time?_

If you want to be comfortable in the terminal, you have to start using the terminal. The easiest way I've found to do this is in-terminal editor (like Vim or Emacs) instead of an IDE.

You could learn about UNIX fundamentals. This project makes heavy use of exit codes, process ids, streams and pipes.

You could learn familiarize yourself with common bash and unix commands. A few are: man, touch, cd, ls, pwd, grep, sed, kill, printf/echo, pwd, rm, test, while, time, &, wait, $!, $$, $?, >, >>, test, ctrl-z/fg

Regardless of the project you end up working on, I can't recommend more highly that you learn a version control system, like git.

- _How can I make my application look good?_

For this project, I don't think its as much that your application is impressive, but that your applications shows your team is the best fit for the project. Its a relatively small project, so it would be great for a team which has little experience in bash/unix, high interest, and good support from a coach with relevant experience.

And no, you absolutely do not have to make any initial contribution to make your applications stand out.

- _What will we be implementing in the summer?_

Everything goes through issues - https://github.com/mweitzel/firetower/issues

-----

And again, please feel free to reply with questions or comments. And don't hesitate to to submit an issue!

**Comment by Codher15** [comment]
Hey
What is required to get the application selected?
Fixing a bug/submitting an issue/a good proposal
Is it necessary to have only experienced members in the team?

**Comment by mweitzel** [comment]
Hi @Codher15, good questions.

- _What is required to get the application selected, fixing a bug/submitting an issue/a good proposal?_

See https://railsgirlssummerofcode.org/students/application/#selection

Fixing a bug or submitting and issue is absolutely **not** a pre-req. But showing how the skill level of your team is appropriate for this project is important.

- _Is it necessary to have only experienced members in the team?_

In short, no. For a RGSoC project, this is probably best suited for a team with a little experience with unix, but high initiative as bash tends to have a steep learning curve. It would be very important for an inexperienced team to have a coach who knows what they're doing.

-----
_edited 2/22/17_

**Comment by codher23** [comment]
Hey
In the present code of firetower, if a null string is entered in the -d or --directory command, then it defaults it to the current directory. We are trying to modify the code so that in case a null string is passed in these arguments as directory then it calls usage and in case of an invalid directory an error message saying "not a directory" is displayed along with call to usage.
If you like these changes, we will open it as an issue and send a PR.

Also, we would like to know some suggestions from your side regarding the features that you are looking forward to be added in firetower.

Moreover, can you tell how many teams you will be selecting this year?

**Comment by mweitzel** [comment]
Closing this ticket as the project was not selected. Thanks everyone for the interest!
